src/core/algorithms/ind/faida/hashing/hashing.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <string>
+
+#include "algorithms/ind/faida/include_lib/murmurhash3/murmur_hash_3.h"
+#include "immintrin.h"
+
+namespace algos::faida::hashing {
+
+inline size_t __attribute__((always_inline)) rotl(size_t hash, int num_bits) {
```

Comment:
It is usually believed that the compiler is smart enough and can handle the inlining of the necessary functions itself. And therefore, declaring a function as inline often will not affect anything.

I would prefer to ask @polyntsov for his opinion on this matter

Reply:
Firstly, `inline` keyword in c++ has nothing to do with function inlining, however, since we want to define this function in the header, `inline` is perfectly fine.
I think that compiler will inline it when compiling with optimizations regardless of presence of `__attribute__((always_inline))` so removing it won't influence performance. At the same time, this is a compiler extension, and IMO it's better not to use compiler extensions when we can (and here we certainly can).
Actually, I think that the best solution is to wait until C++ version is bumped to c++20 and use `std::rotl` here